ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js

```diff
@@ -606,7 +607,12 @@ if (typeof window.DatePickerWidget === 'undefined') {
       this.#dp.style.left = styles.left;
       const localeObj = new Intl.Locale(this.#lang);
       if(localeObj?.textInfo?.direction == "rtl") {
-        this.#dp.style.right = styles.left;
+        let right = windowInnerWidth - (left + inputRect.width); // Calculate right offset
+        if (right + 433 > windowInnerWidth) { // 433 is the widget's width
```

Comment:
hardcoding the widget's width might not be appropriate across different viewports. Can we calculate the widget's width based on the actual element dimensions.